### ISSUE TITLE
Go To Declaration fails with aliasing

### ISSUE THREAD

**Issue by William65536** [zigtools/zls#2054]
### Zig Version

0.14.0-dev.1511+54b668f8a

### ZLS Version

0.14.0-dev.147+dd78968

### Client / Code Editor / Extensions

VS Code

### Steps to Reproduce and Observed Behavior

If you use Go To Declaration on `baz` ZLS jumps you to `foo`.

```lang-zig
const foo = 0;
const bar = foo;
const baz = bar; // Go To Declaration on baz jumps to `foo`
```

### Expected Behavior

Go To Declaration should respect aliasing and take you to `bar` instead of `foo`. This works correctly if you Go To Declaration on `bar` instead:

```lang-zig
const foo = 0;
const bar = foo;
const baz = bar; // Go To Declaration on bar correctly jumps to `const bar = ...`
```

**Comment by xdBronch** [comment]
are you sure youre using goto declaration and not goto definition? this is working as expected for me

**Comment by William65536** [comment]
Yes. The behavior is the same for go to definition and go to declaration with the `bar` case. Perhaps this has been fixed in a more recent version.

**Comment by Tesseract22** [comment]
This does not work for me as well. 
zls commit: 8406f8c7789dcf8363b972ed306535ac4ec80e71
zig version: 0.14.0-dev.2571+01081cc8e
